Source/Core/InputCommon/ControllerInterface/OctagonalMouseJail.cpp

```diff
@@ -0,0 +1,385 @@
+    return;
+  }
+
+  long index_of_second_line_point =
+      FindSecondLinePoint(mouse_point, index_of_closest_octagon_point);
+
+  PrimaryPrimitive rise = m_octagon_points[index_of_second_line_point].y -
+                          m_octagon_points[index_of_closest_octagon_point].y;
+  PrimaryPrimitive run = m_octagon_points[index_of_second_line_point].x -
+                         m_octagon_points[index_of_closest_octagon_point].x;
+  PrimaryPrimitive slope_of_octagon_line = rise / run;
+  PrimaryPrimitive y_intercept_of_octagon_line =
+      slope_of_octagon_line * (0 - m_octagon_points[index_of_closest_octagon_point].x) +
+      m_octagon_points[index_of_closest_octagon_point].y;
+
+  rise = m_render_window.center.y - mouse_point.y;
+  run = m_render_window.center.x - mouse_point.x;
+  PrimaryPrimitive slope_of_line_from_center_to_mouse = rise / run;
+  constexpr char distance_from_y_axis_where_math_breaks_down = 1.0;
+  if (std::abs(run) < distance_from_y_axis_where_math_breaks_down)
+  {
+    if (mouse_point.y < m_render_window.center.y)
+    {
+      mouse_point = m_octagon_points[NORTH];
+    }
+    if (mouse_point.y > m_render_window.center.y)
+    {
+      mouse_point = m_octagon_points[SOUTH];
+    }
+    return;
+  }
+  PrimaryPrimitive y_intercept_of_line_from_center_to_mouse =
+      slope_of_line_from_center_to_mouse * (0 - m_render_window.center.x) +
+      m_render_window.center.y;
+
+  Point point_of_intersection = {};
+
+  point_of_intersection.x =
+      std::round(((y_intercept_of_octagon_line - y_intercept_of_line_from_center_to_mouse) /
+                  (slope_of_line_from_center_to_mouse - slope_of_octagon_line)));
+  point_of_intersection.y =
+      std::round((slope_of_octagon_line * point_of_intersection.x) + y_intercept_of_octagon_line);
+
+  mouse_point = point_of_intersection;
+}
+
+void OctagonalMouseJail::UpdateRenderWindowInfo()
+{
+  WindowInfo render_window_info = Host_GetRenderWindowInfo();
+  m_render_window.handle = render_window_info.handle;
+  m_render_window.top = render_window_info.top;
+  m_render_window.left = render_window_info.left;
+  m_render_window.bottom = render_window_info.bottom;
+  m_render_window.right = render_window_info.right;
+  m_render_window.width = m_render_window.right - m_render_window.left;
+  m_render_window.height = m_render_window.bottom - m_render_window.top;
+  m_render_window.center.y = (m_render_window.height / 2.0) + m_render_window.top;
+  m_render_window.center.x = (m_render_window.width / 2.0) + m_render_window.left;
+  if (m_render_window.height == 0)
+    m_render_window.ratio = 0;
+  else
+    m_render_window.ratio = m_render_window.width / m_render_window.height;
+  if (m_render_window.handle == nullptr)
+    m_octagon_points = std::array<Point, 8>{};
+  else
+    m_octagon_points = GenerateOctagonPoints();
+}
+
+bool OctagonalMouseJail::IsPlayerRequestingCenter()
+{
+  return HotkeyManagerEmu::IsPressed(Hotkey::HK_CENTER_MOUSE, true);
```

Comment:
I'm not sure why this doesn't link on nogui, but it is clearly different than how all other hotkeys are implemented. Might be best to just follow the standard path there.

Reply:
I need this to be here or else the key doesn't function the way I want it to. It doesn't center the mouse continuously if it's implemented like the other keys are.

Reply:
The nogui failure was (fingers crossed) me forgetting to include `HotkeyManager.h' in my file.

Reply:
Well you either need to figure out why this doesn't link and also verify that IsPressed() is safe to call from anywhere (it looks race-y to me but I'm not very familiar with this code, to be fair), or we need a different solution for this. I suspect something like what the `HK_TOGGLE_THROTTLE` hotkey does could work here.